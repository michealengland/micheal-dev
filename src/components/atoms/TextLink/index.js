import React from 'react'
import Link from 'next/link'
import PropTypes from 'prop-types'

export default function TextLink({children, to, activeClassName, partiallyActive, ...other}) {
  return (
    <Link
      // TODO FIX props on this.
      // activeClassName={activeClassName}
      // partiallyActive={partiallyActive}
      href={to}
      {...other}
    >
      {children}
    </Link>
  )
}

TextLink.propTypes = {
  activeClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
  partiallyActive: PropTypes.bool,
  to: PropTypes.string.isRequired,
}
