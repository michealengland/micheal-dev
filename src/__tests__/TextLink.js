import React, {Fragment} from 'react'
import renderer from 'react-test-renderer'
import TextLink from '../components/atoms/TextLink'
describe('TextLink', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Fragment>
          <TextLink to="/">Home Link</TextLink>
          <TextLink to="https://micheal.dev">Home Link</TextLink>
        </Fragment>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
