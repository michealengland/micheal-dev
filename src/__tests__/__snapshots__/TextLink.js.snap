// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink renders correctly 1`] = `
Array [
  <a
    href="/"
  >
    Home Link
  </a>,
  <a
    href="https://micheal.dev"
  >
    Home Link
  </a>,
]
`;
